core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -6562,6 +6562,67 @@ private HepProgram getTransitiveProgram() {
         .check();
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6214">[CALCITE-6214]
+   * Remove `DISTINCT` in `COUNT` if field is unique</a>. */
+  @Test void testAggregateDistinctRemove1() {
+    final String sql = ""
+        + "select count(distinct x) cnt\n"
+        + "from(\n"
+        + "   select distinct sal x from emp\n"
+        + ") t ";
+    sql(sql)
+        .withRule(CoreRules.AGGREGATE_REMOVE_DISTINCT)
+        .check();
+  }
+
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6214">[CALCITE-6214]
+   * Remove `DISTINCT` in `COUNT` if field is unique</a>. */
+  @Test void testAggregateDistinctRemove2() {
+    final String sql = ""
```

Comment:
Can you add at least one test where the outer query has a `GROUP BY`? The following query should benefit from the simplification:
```
SELECT deptno, COUNT(DISTINCT sal), 
FROM (
   SELECT DISTINCT deptno, sal
   FROM emp)
GROUP BY deptno
```
Note that `sal` is not distinct but it is distinct for each `deptno` (because `(deptno, sal)` is a key).

Reply:
Added it.